### DISCUSSION TITLE
issue: some errors are not cleared or displayed after updating dependent form fields (Yup resolver)

### DISCUSSION THREAD

**Discussion by fwolff** [react-hook-form/react-hook-form#7692] [Q&A] [answered]
### Version Number

7.25.1

### Codesandbox/Expo snack

https://codesandbox.io/s/compassionate-rgb-kztu5?file=/src/App.tsx

### Steps to reproduce

1. Go to [sandbox link](https://codesandbox.io/s/compassionate-rgb-kztu5?file=/src/App.tsx)
2. Enter 'ab' in the first text field.
3. Enter '2' in the second text field.
4. Displayed error is 'age must be greater than or equal to 5' (OK)
5. Append 'c' to 'ab' in the first text field.
6. Displayed error is still 'age must be greater than or equal to 5' (KO)

### Expected behaviour

Error message should be cleared after step 5.

When debugging the onChange method in react-hook-form/src/logic/createFormControl.ts, I can see (line https://github.com/react-hook-form/react-hook-form/blob/954be6d190548c9fc0f83017d7b0f5f4786d0a64/src/logic/createFormControl.ts#L712) that my Yup schema is evaluated and that the error should be cleared.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment, marked as answer]
You might wanna look into `deps` option for `register` form method. See [the RHF docs section on the API](https://react-hook-form.com/api/useform/register)
```tsx
{...register("firstName", { deps: ["age"] })}
```
[Working CSB example](https://codesandbox.io/s/elegant-lederberg-eird5?file=/src/App.tsx:1086-1103)

**Comment by bluebill1049** [comment]
hook form focus on field level, for depdent validation use `trigger`.

https://react-hook-form.com/api/useform

> Schema validation focus on the field level for error reporting. Parent level error look is only limited to the direct parent level that is applicable for components such as group checkboxes.
